packages/editor/src/components/media-placeholder/index.js

```diff
@@ -234,4 +257,18 @@ class MediaPlaceholder extends Component {
 	}
 }
 
-export default withFilters( 'editor.MediaPlaceholder' )( MediaPlaceholder );
+const applyWithSelect = withSelect( ( select ) => {
+	let hasUploadPermissions = false;
+	if ( undefined !== select( 'core' ) ) {
```

Comment:
I think this check is not necessary, to be consistent with other `withSelect` usage, we should assume the "core" store to be available as it's a dependency of the editor package.

Reply:
I think I saw in the comments that it's because of the unit tests. Two suggestions:

 - Avoid running this code in unit tests (test the inner component)
 - Import the `@wordpress/core-data` in the unit tests to ensure that the store is available.